source/server/server.h

```diff
@@ -160,6 +163,18 @@ class ServerFactoryContextImpl : public Configuration::ServerFactoryContext {
   TimeSource& timeSource() override { return api().timeSource(); }
   Api::Api& api() override { return server_.api(); }
 
+  // Configuration::TransportSocketFactoryContext
+  Ssl::ContextManager& sslContextManager() override { return server_.sslContextManager(); }
+  Secret::SecretManager& secretManager() override { return server_.secretManager(); }
+  Stats::Store& stats() override { return server_.stats(); }
+  void setInitManager(Init::Manager&) override {}
+  Init::Manager* initManager() override { return &server_.initManager(); }
+  ProtobufMessage::ValidationVisitor& messageValidationVisitor() override {
+    return initManager()->state() == Init::Manager::State::Initialized
```

Comment:
Why initialization matters?

Reply:
`server_` has two validation visitors (dynamic and static), so returning just one of them would restrict the use of the returned transport socket factory context to either dynamic (e.g., xDS, after bootstrap) or static (e.g., bootstrap) configuration context. Choosing the visitor based on the initialization state allows the added callback can be used for both static and dynamic configuration. In `server/server.cc` the validation visitor is selected based on whether the function in question is doing dynamic or static configuration. I'll add a comment explaining this.

Reply:
I admit the observation is generally right but the foundation is very fragile. A server factory context definitely can be used to handle dynamic config when bootstrap is not yet done, and vice versa.

It probably works correctly in your call it since it belongs to `TransportSocketFactoryContext`